### ISSUE TITLE
Kibana 6.4.0 support

### ISSUE THREAD

**Issue by ppadovani** [ppadovani/KibanaNestedSupportPlugin#94]
Support Kibana 6.4.0

**Comment by ppadovani** [comment]
Quick Update: First pass shows changes to many imports used by this plugin. Additionally, the changes from 6.3 to 6.4 to code this plugin replaces/updates were not insignificant. This port will take some time.

**Comment by ppadovani** [comment]
Houston we may have a problem. The Kibana team has changed the export declaration on a bunch of the objects I override/inject code into from

```
export something = new ObjectName() {
...}
```

to:

```
export const something = new ObjectName() {
...}
```

This may mean that I cannot override/inject my code. If this is the case, then I will have to spend significant resources/time trying to find other ways to do what needs to be done. In the end I may not be able to. I'll update this issue again within a week or so.

**Comment by ppadovani** [comment]
Still working on this....

**Comment by ppadovani** [comment]
My apologies on the silence. Work and life took priority over this work. I will be restarting this effort this weekend.

**Comment by ppadovani** [comment]
Quick update:
  - Index management is done.
  - Discover app: functional minus syntax error display
  - Vis.: still broken.

I am cautiously optimistic that I'll have a release ready by next weekend.

**Comment by mxcoder** [comment]
Thanks so much!
Really appreciate all your work for this.

**Comment by ppadovani** [comment]
Update: Got everything working except the Metric visualization. Unfortunately higher priorities in regards to work have intruded. I will likely not release this until next weekend.

**Comment by ppadovani** [comment]
This work stalled due to work commitments. All of the code is pushed to the 6.4 branch, and I just need to finish fixing the metric and final testing. I hope to get back to this in the next week or so.

**Comment by ppadovani** [comment]
@mxcoder @mackermann2 @ngsandbox @onlyway
I've got an initial release candidate out, please pull and test and let me know in this issue anything that is broken.

https://github.com/ppadovani/KibanaNestedSupportPlugin/releases/download/6.4.0-1.0.0-rc1/nested-fields-support-6.4.0-1.0.0-rc1.zip

**Comment by mackermann2** [comment]
Hello Pierre, 
Thanks for this new release. Unfortunately we are using the next version 6.4.1 of Kibana.
So I have this error during the plugin installation : 

_[root@kibana ~]# /usr/share/kibana/bin/kibana-plugin install https://github.com/ppadovani/KibanaNestedSupportPlugin/releases/download/6.4.0-1.0.0-rc1/nested-fields-support-6.4.0-1.0.0-rc1.zip
Attempting to transfer from https://github.com/ppadovani/KibanaNestedSupportPlugin/releases/download/6.4.0-1.0.0-rc1/nested-fields-support-6.4.0-1.0.0-rc1.zip
Transferring 261422 bytes....................
Transfer complete
Retrieving metadata from plugin archive
Extracting plugin archive
Extraction complete
**Plugin installation was unsuccessful due to error "Incorrect Kibana version in plugin [nested-fields-support]. Expected [6.4.1]; found [6.4.0]"**_

Do you think I can just modify the number of version in "package.json" file ? 
Or is it better to downgrade our version to 6.4.0 ? 

Regards,
Matthieu

**Comment by ppadovani** [comment]
I will build a 6.4.1-1.0.0-rc1 candidate this weekend. It shouldn't take much effort.

@mackermann2 changing the version in the package.json file should work, but may expose issues not seen in 6.4.0, which you can report here.

**Comment by ppadovani** [comment]
Released 6.4.0, 6.4.1 and 6.4.2
